src/renderer/coremods/devCompanion/index.tsx

```diff
@@ -0,0 +1,210 @@
+import { Logger } from "../../modules/logger";
+import {
+  getByProps,
+  getBySource,
+  getByValue,
+  getModule,
+  sourceStrings,
+} from "src/renderer/modules/webpack";
+import { Filter } from "src/types";
+
+const PORT = 8485;
+
+const devLogger = Logger.api("DevCompanion");
+
+export let socket: WebSocket | undefined;
+
+type Node = StringNode | RegexNode | FunctionNode;
+
+interface StringNode {
+  type: "string";
+  value: string;
+}
+
+interface RegexNode {
+  type: "regex";
+  value: {
+    pattern: string;
+    flags: string;
+  };
+}
+
+interface FunctionNode {
+  type: "function";
+  value: string;
+}
+
+interface PatchData {
+  find: string | RegExp;
+  replacement: Array<{
+    match: StringNode | RegexNode;
+    replace: StringNode | FunctionNode;
+  }>;
+}
+
+interface FindData {
+  type: string;
+  args: Array<StringNode | FunctionNode>;
+}
+
+function parseNode(node: Node): string | RegExp {
+  switch (node.type) {
+    case "string":
+      return node.value;
+    case "regex":
+      return new RegExp(node.value.pattern, node.value.flags);
+    case "function":
+      // We LOVE remote code execution
+      // Safety: This comes from localhost only, which actually means we have less permissions than the source,
+      // since we're running in the browser sandbox, whereas the sender has host access
+      // eslint-disable-next-line no-eval
+      return (0, eval)(node.value);
+    default:
+      throw new Error(`Unknown Node Type ${(node as Node).type}`);
+  }
+}
+
+function parseFind(type: string, args: unknown[]): unknown {
+  console.log(`Received find parsing request of type ${type} with args: ${args}`);
+  switch (type.replace("get", "")) {
+    case "Module":
+      return getModule(args[0] as Filter, { all: true });
+    case "ByProps":
+      return getByProps(args as string[], { all: true });
+    case "BySource":
+      return getBySource(args[0] as string | RegExp, { all: true });
+    case "ByValue":
+      return getByValue(args[0] as string | RegExp, { all: true });
+    default:
+      throw new Error(`Unknown Find Type ${type}`);
+  }
+}
+
+function search(find: string | RegExp): string[] {
+  return Object.values(sourceStrings).filter((s) => {
+    if (typeof find === "string") {
+      return s.includes(find);
+    }
+    if (find instanceof RegExp) {
+      return find.test(s);
```

Comment:
if the regex is global, this will have unexpected results because global regexes keep state for test and exec. either use something like `!!s.match(find)` or reset the regex:
```suggestion
      const matches = find.test(s);
      find.lastIndex = 0;
      return matches;
```